lua/indentmini/init.lua

```diff
@@ -34,7 +34,24 @@ local function indentline(hl_current)
 
     ctx[row] = indent
 
-    for i = 1, indent - 1, vim.bo[bufnr].sw do
+    local function indent_step()
+      if vim.fn.exists('*shiftwidth') ~= 0 then
+        return vim.fn.shiftwidth()
+      elseif vim.fn.exists('+shiftwidth') ~= 0 then
+        -- sample impl of shiftwidth builtin
+        if vim.opt_local.shiftwidth:get() ~= 0 then
```

Comment:
I don't think there need check 0 if 0 nothing happen in for loop. so indent line not set .

tabstop check in there a bit wired. it use for expandtab set to true. but the indent still use shiftwdith

Reply:
If the value of `shiftwidth` .ie shift_step is `0`, the loop runs forever. The issue is that there are cases where shiftwidth is `0` even though there are indents in the file. This is why I had this issue in the first place. The `+shiftwidth` checks for the shiftwidth option, and tries to prevent the `0` case issue. I based the implementation on the documentation of fn.shiftwidth.

Reply:
But I will do the change to check for 1 instead of not `0`, since that seems to be clearer

Reply:
> If the value of `shiftwidth` .ie shift_step is `0`, the loop runs forever. The issue is that there are cases where shiftwidth is `0` even though there are indents in the file. This is why I had this issue in the first place. `fn.shiftwidth()` perfectly fixes this issue, but since we are checking its existence, assuming a case where it doesn't exist, we use the impl in the `elseif` branch. The `+shiftwidth` checks for the shiftwidth option, and tries to prevent the `0` case issue. I based the implementation on the documentation of `fn.shiftwidth`, which also uses tabstop when shiftwidth is 0.

This is an explanation of why that `tabstop` is also used.

Reply:
When editor_config is used for configuring basic options in nvim, [check indent_size and indent_style section ](https://neovim.io/doc/user/editorconfig.html) and its [code in nvim](https://github.com/neovim/neovim/blob/63b3408551561127f7845470eb51404bcd6f547b/runtime/lua/editorconfig.lua#L49). There are cases where the `shiftwidth` option is `0`. In this case, `tabstop` is used as specified by [shiftwidth docs](https://neovim.io/doc/user/builtin.html#shiftwidth()). So the else branch just emulates the implementation of `fn.shiftwidth`